pkg/codegen/go/gen_program_expressions.go

```diff
@@ -373,8 +375,21 @@ func (g *generator) genScopeTraversalExpression(w io.Writer, expr *model.ScopeTr
 	if isInput {
 		g.Fgenf(w, "%s(", argumentTypeName(expr, expr.Type(), isInput))
 	}
-	g.Fgen(w, rootName)
-	g.genRelativeTraversal(w, expr.Traversal.SimpleSplit().Rel, expr.Parts, objType)
+
+	if rootName == "range" {
+		part := expr.Traversal[1].(hcl.TraverseAttr).Name
+		switch part {
+		case "value":
+			g.Fgenf(w, "val0")
+		case "key":
+			g.Fgenf(w, "key0")
+		default:
+			contract.Failf("unexpected traversal on range expression: %s", part)
+		}
+	} else {
```

Comment:
I'm not sure about this. There could be other variables named `range` besides the range expression itself. The other generators solve this by emitting a variable inside the scope of the ranged resource named `range` with the properties `key` and `value`. I do think that this will probably suffice for the TF fronted, though, so no rush.

Reply:
Ah, I see. Yeah, this could lead to conflicts. For the time being, I will go ahead and add a TODO to replace this code with a function call expression that uses range/entries.